pkg/pillar/hypervisor/xen.go

```diff
@@ -49,6 +50,61 @@ type xenContext struct {
 	ctrdContext
 }
 
+func (ctx xenContext) GetCapabilities() (types.Capabilities, error) {
+	errorString := ""
+	vtx, err := ctx.checkVTX()
+	if err != nil {
+		errorString = err.Error()
+	}
+	vtd, err := ctx.checkVTD()
+	if err != nil {
+		errorString = errorString + "; " + err.Error()
+	}
+	if errorString != "" {
+		err = fmt.Errorf("GetCapabilities: %s", errorString)
+	} else {
+		err = nil
+	}
+	return types.Capabilities{
+		Hypervisor: ctx.Name(),
+		VTX:        vtx,
+		VTD:        vtd,
+	}, err
+}
+
+func (ctx xenContext) checkVTX() (bool, error) {
+	switch runtime.GOARCH {
+	case "arm64":
+		//xen requires extensions to run on ARM64
```

Comment:
Not sure I understand what this means

Reply:
In my experiments with arm64 (`-machine virt` and `-machine virt,virtualization=true`) I cannot run Xen without Arm virtualization extensions. Do you have any suggestions here?

Reply:
That actually relates to my very first question: what do you want these fields to mean. We can agree that they mean abstract notions of H/W assisted virtualization and I/O virtualization OR we can make them arch specific.